src/rebar_git_resource.erl

```diff
@@ -188,6 +188,55 @@ maybe_warn_local_url(Url) ->
         _ -> ok
     end.
 
+git_cache_dir(Url, Tag) ->
+    case os:getenv("REBAR_GIT_CACHE_DIR") of
+        false ->
+            nocache;
+        Dir ->
+            Full = case parse_git_url_as_cache_path(Url) of
+                nocache ->
+                    nocache;
+                "/" ++ Path ->
+                    filename:join([Dir, Path, Tag])
+            end,
+            DotGit = filename:join(Full, ".git"),
+            case filelib:is_dir(DotGit) of
+                true ->
+                    {hit, Full};
+                false ->
+                    {miss, Full}
+            end
+    end.
+
+parse_git_url_as_cache_path(Url) ->
+    case parse_git_url(Url) of
+        {ok, {Host, Path}} ->
+            filename:join([Host, Path]);
+        Other ->
+            ?DEBUG("Failed to parse url ~ts ~p", [Url, Other]),
+            nocache
+    end.
+
+git_clone2(tag, GitVsn, Url, Dir, Tag) ->
+    CloneF = fun() -> git_clone(tag, GitVsn, Url, Dir, Tag) end,
+    case git_cache_dir(Url, Tag) of
+        nocache ->
+            ok = CloneF();
+        {hit, CacheDir} ->
+            %% cache hit, copy from the cache
+            ?DEBUG("Git cache hit: ~ts", [CacheDir]),
+            rebar_file_utils:cp_r([CacheDir], Dir);
+        {miss, CacheDir} ->
+            %% cache miss, cache it so it can be used next time
+            ok = CloneF(),
```

Comment:
Another option is to not autopopulate the cache here, which will also make the problem above irrelevant.
We could instead update the cache periodically via a separate single process.

Reply:
Added `REBAR_GIT_CACHE_REF_AUTOFILL`.
Set it to `0` to avoid cache auto-fill.